convert_old_db/convert.cc

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2016 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#include <cassert>
+#include <fstream>
+
+#include "new_schema.h"
+#include "new_storage.h"
+
+#include "../../Current/Bricks/dflags/dflags.h"
+#include "../../Current/Bricks/time/chrono.h"
+
+#include "../../Current/TypeSystem/struct.h"
+#include "../../Current/TypeSystem/Serialization/json.h"
+
+DEFINE_string(input, "db.json", "The name of the input data to convert.");
+DEFINE_string(output, "new_db.json", "The name of the input data to convert.");
+
+// TODO(dkorolev) + TODO(mzhurovich): Should we expose `T_FIELDS_VARIANT` or `T_TRANSACTION` from Storage?
```

Comment:
`T_TRANSACTION` is public in Storage and `T_FIELDS_VARIANT` is private. We could make it public as well.